10_rest/willshen8/data/invalid_puppies.json

```diff
@@ -0,0 +1,5 @@
+[
```

Comment:
I understand, but this is inherited from previous labs where the test cases were mainly for the flag `--data` where you had to pass in a JSON file. I could change the test cases to be `io.reader`, but there is little to be gained.

Reply:
well lucky you that this slipped through in a previous lab, however I'm reviewing all code here and to me this is not ideal ;) 

i can see that the lab-setup has got it's frustrations - on the other side you get the same code re-reviewed a few times and get a chance to really get it to be great/many people's input.  
to me depending on files is something pretty important to be avoided as much as possible because it makes tests harder to read (files are not local to the test case as a string used in a strings.Reader would be) and you also have to be careful with paths/ from where you execute your tests etc - in short they make tests more fragile and less readable and seem like a poor abstraction compared with reader/writer.

this is the last lab to finish the course, so I'd strongly encourage you to put in the last little bit of effort ;) i think it will make us both feel better at the end.

(there will be time to do it all).